### ISSUE TITLE
Proxy causes url rewrite in XML response of WMS

### ISSUE THREAD

**Issue by fzadrazil** [Rob--W/cors-anywhere#353]
When using web mapping services (WMS), first request goes to the GetCapabilities method that returns XML description of the service. There are urls inside this XML which are supposed to link to the original server, but when proxyfied with cors-anywhere, those urls gets rewritten. I am not sure whether the proxy itself can somehow rewrite those or it just fails to propagete correct hostname so the WMS server constructs flawed XML based on the wrong host name.

See eg. (the url in this XML should still be at the mapy.geology.cz domain, but has the domain of the proxy instead)
https://hub.lesprojekt.cz/proxy/https://mapy.geology.cz/arcgis/services/Dulni_Dila/poddolovana_uzemi/MapServer/WMSServer?request=GetCapabilities&service=WMS&version=1.3.0
![image](https://user-images.githubusercontent.com/48727502/116563509-625a9900-a904-11eb-8dc3-8f20b1a5a0ac.png)

Other proxies we used don't do that.

**Comment by Rob--W** [comment]
CORS Anywhere does not rewrite the response bodies to requests. If the response is different, then either your destination server is serving different responses to the other proxies, or it is serving a different response to CORS Anywhere (or the IP of the demo server).

You can check whether it's specific to the IP addresses of CORS Anywhere or the shape of the request by (temporarily) hosting it on your own server (or localhost) and checking whether the response still differs.

**Comment by fzadrazil** [comment]
#response_container_BBPPID{font-family: initial; font-size:initial; color: initial;}  I have already tried on multiple servers and with multiple WMS providers. The issue remains for all and even for CORS anywhere on localhost.                                                                                                                                                                                                                                                                                                                         Od: ***@***.***Odesláno: 29. dubna 2021 4:46 odpoledneKomu: ***@***.***Odpověď komu: ***@***.***Kopie: ***@***.***; ***@***.***Předmět: Re: [Rob--W/cors-anywhere] Proxy causes url rewrite in XML response of WMS (#353)  
CORS Anywhere does not rewrite the response bodies to requests. If the response is different, then either your destination server is serving different responses to the other proxies, or it is serving a different response to CORS Anywhere (or the IP of the demo server).
You can check whether it's specific to the IP addresses of CORS Anywhere or the shape of the request by (temporarily) hosting it on your own server (or localhost) and checking whether the response still differs.

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by Rob--W** [comment]
Again, CORS Anywhere doesn't rewrite response bodies. So this is an issue with your destination server. You could try sending a request to a URL that returns the response headers to check whether the difference is caused by differences in request headers.

**Comment by fzadrazil** [comment]
> CORS Anywhere doesn't rewrite response bodies

Yes, I supposed that is the case as there is not reason for it. I just needed to check. 

> So this is an issue with your destination server.

It is not our server, but widely used mapping services by multiple providers. Not really any chance to change anything on the service side. But we have found out that it probably uses the X-Forwarded-Host header to construct those URLs. So I digged deep into the CORS Anywhere documentation and found that those headers can be turned off in http-proxy. Based on that I have set 

```
  httpProxyOptions: {
    xfwd: false,            // Append X-Forwarded-* headers
  }
```
and seems to work at least on localhost. It still does the same when deployed on the server though, but I guess it is because those headers are added somewhere else in the infrastructure.
